src/main/java/ca/team2706/frc/robot/commands/drivebase/DriverAssistVision.java

```diff
@@ -14,189 +15,335 @@
      */
-    private double distanceCameraToTarget_CameraPrev = 0.0;
+    private NetworkTableEntry driverEntry;
 
-    /**
-     * True if target is cargo ship or loading bay
+    /*
+     * Network table entry named "Cargo", a boolean set to true to request to the vision
+     * to detect ball targets, false otherwise
      */
-    private boolean driverAssistCargoAndLoading = false;
+    private NetworkTableEntry findCargoEntry;
 
-    /**
-     * True if target is rocket
+    /*
+     * Network table entry named "Tape", a boolean set to true to request to the vision
+     * to detect tape targets, false otherwise
      */
-    private boolean driverAssistRocket = false;
+    private NetworkTableEntry findTapeEntry;
 
-    /**
+    /*
+     * Motion control system object providing functionality to follow a Pathfinder trajectory
+     */
+    private FollowTrajectory followTrajectory;
+
+    /*
+     * True if the generateTrajectoryRequest stage of execute() method has completed,
+     * indicating that a command has been issued to generate a trajectory, false otherwise
+     */
+    boolean generateTrajectoryRequestStageComplete = false;
+
+    /*
      * Network table instance to get data from vision subsystem
      */
     private NetworkTableInstance inst;
 
     /**
-     * Network table to get data from vision subsystem
+     * Network table to get Pathfinder data from vision subsystem
      */
-    private NetworkTable table;
+    private NetworkTable pathfinderTable;
 
     /**
-     * Trajectory generated to move robot to target
+     * True if ring light on stage of execute() method has completed, false otherwise
      */
-    private Trajectory traj;
+    private boolean ringLightOnStageComplete = false;
 
     /**
-     * True if trajectory has been generated for this command, false otherwise.
-     * (This will be removed when motion control functionality is implemented.)
+     * Current value of time delay after ring light has been turned on.
+     */
+    private double ringLightOnDelayTime = 0.0;
+
+    /*
+     * Network table entry named TapeDetected, a boolean set to true if the tape target has been
+     * detected by the vision system, false otherwise
+     */
+    private NetworkTableEntry tapeDetectedEntry;
+
+    /*
+     * Type of target that robot will move to (CARGO_AND_LOADING, ROCKET, or BALL)
      */
-    private boolean trajGenerated = false;
+    private DriverAssistVisionTarget target;
 
     /**
-     * True if vision system has been assessed to be offline
+     * Trajectory generated by Pathfinder to move robot to target. Declared volatile since it is shared between
+     * different threads.
      */
-    boolean visionOffline = false;
+    private volatile Trajectory trajectory;
 
     /**
      * Creates empty driver assist command object (needed for unit testing framework only)
      */
     public DriverAssistVision() {
         // Ensure that this command is the only one to run on the drive base
```

Comment:
This comment should be updated to reflect the change in code

Reply:
Updated.

Reply:
Actually, I just found a way to get rid of the DriverAssistVision() constructor that was there purely for test purposes. (Used Injectable the way it was used in other test classes.)